File: src/pattern/abstractFactory/listFactory/java/ListFactory.java
package pattern.abstractFactory.listFactory.java;

import pattern.abstractFactory.factory.java.Factory;
import pattern.abstractFactory.factory.java.Link;
import pattern.abstractFactory.factory.java.Page;
import pattern.abstractFactory.factory.java.Tray;

public class ListFactory extends Factory {
    @Override
    public Link createLink(String caption, String url) {
        return new ListLink(caption, url);
    }

    @Override
    public Tray createTray(String caption) {
        return new ListTray(caption);
    }

    @Override
    public Page createPage(String title, String author) {
        return new ListPage(title, author);
    }
}

File: src/pattern/abstractFactory/listFactory/kotlin/ListFactory.kt
package pattern.abstractFactory.listFactory.kotlin

import pattern.abstractFactory.factory.kotlin.Factory
import pattern.abstractFactory.factory.kotlin.Item
import pattern.abstractFactory.factory.kotlin.Page
import pattern.abstractFactory.factory.kotlin.Tray

class ListFactory: Factory() {
    override fun createLink(caption: String, url: String): Item {
        return ListLink(caption, url)
    }

    override fun createTray(caption: String): Tray {
        return ListTray(caption)
    }

    override fun createPage(title: String, author: String): Page {
        return ListPage(title, author)
    }


}